vault/resource_pki_secret_backend_cert.go

```diff
@@ -0,0 +1,209 @@
+				Default:      "der",
+				ValidateFunc: validation.StringInSlice([]string{"der", "pkcs8"}, false),
+			},
+			"exclude_cn_from_sans": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Flag to exclude CN from SANs.",
+				ForceNew:    true,
+			},
+			"certificate": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The certicate.",
+			},
+			"issuing_ca": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The issuing CA.",
+			},
+			"ca_chain": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The CA chain.",
+			},
+			"private_key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The private key.",
+			},
+			"private_key_type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The private key type.",
+			},
+			"serial_number": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The serial number.",
+			},
+		},
+	}
+}
+
+func pkiSecretBackendCertCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	name := d.Get("name").(string)
+
+	path := pkiSecretBackendCertPath(backend, name)
+
+	commonName := d.Get("common_name").(string)
+
+	iAltNames := d.Get("alt_names").([]interface{})
+	altNames := make([]string, 0, len(iAltNames))
+	for _, iAltName := range iAltNames {
+		altNames = append(altNames, iAltName.(string))
+	}
+
+	iIPSans := d.Get("ip_sans").([]interface{})
+	ipSans := make([]string, 0, len(iIPSans))
+	for _, iIpSan := range iIPSans {
+		ipSans = append(ipSans, iIpSan.(string))
+	}
+
+	iOtherSans := d.Get("other_sans").([]interface{})
+	otherSans := make([]string, 0, len(iOtherSans))
+	for _, iOtherSan := range iOtherSans {
+		otherSans = append(otherSans, iOtherSan.(string))
+	}
+
+	data := map[string]interface{}{
+		"common_name":          d.Get("common_name").(string),
+		"ttl":                  d.Get("ttl").(int),
+		"format":               d.Get("format").(string),
+		"private_key_format":   d.Get("private_key_format").(string),
+		"exclude_cn_from_sans": d.Get("exclude_cn_from_sans").(string),
+	}
+
+	if len(altNames) > 0 {
+		data["alt_names"] = altNames
+	}
+
+	if len(ipSans) > 0 {
+		data["ip_sans"] = ipSans
+	}
+
+	if len(otherSans) > 0 {
+		data["other_sans"] = otherSans
+	}
+
+	log.Printf("[DEBUG] Creating certificate %s by %s on PKI secret backend %q", commonName, name, backend)
+	resp, err := client.Logical().Write(path, data)
+	if err != nil {
+		return fmt.Errorf("error creating certificate %s by %s for PKI secret backend %q: %s", commonName, name,
+			backend, err)
+	}
+	log.Printf("[DEBUG] Created certificate %s by %s on PKI secret backend %q", commonName, name, backend)
+
+	d.Set("certificate", resp.Data["certificate"])
+	d.Set("issuing_ca", resp.Data["issuing_ca"])
+	d.Set("ca_chain", resp.Data["ca_chain"])
```

Comment:
Hi @bhuisgen !

The value ca_chain returned by the API is a slice of string: https://www.vaultproject.io/api/secret/pki/index.html#sample-response-8. So with the actual code, the ca_chain will always be null. I don't know if Terraform already provides some util functions to do the conversion but here is mine if you want to use it to convert the slice of interface into a slice of string:

```hcl

d.Set("ca_chain", strings.Join(convertIntoSliceOfString(resp.Data["ca_chain"])[:], "\n"))

func convertIntoSliceOfString(slice interface{}) []string {
	intSlice := slice.([]interface{})
	strSlice := make([]string, len(intSlice))
	for i, v := range intSlice {
		strSlice[i] = v.(string)
	}
	return strSlice
}
```